hawkbit-repository/hawkbit-repository-test/src/main/java/org/eclipse/hawkbit/repository/test/util/MsSqlTestDatabase.java

```diff
@@ -8,67 +8,57 @@
  */
 package org.eclipse.hawkbit.repository.test.util;
 
-import java.sql.Connection;
-import java.sql.DriverManager;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-
 import org.apache.commons.lang3.RandomStringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.test.context.TestExecutionListener;
+import org.springframework.util.StringUtils;
 
 /**
  * A {@link TestExecutionListener} for creating and dropping MS SQL Server
  * schemas if tests are setup with MS SQL Server.
  */
 public class MsSqlTestDatabase extends AbstractSqlTestDatabase {
 
-    private static final Logger LOG = LoggerFactory.getLogger(MsSqlTestDatabase.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(MsSqlTestDatabase.class);
+    private static final String MSSQL_URI_PATTERN = "jdbc:sqlserver://{host}:{port};database={db}*";
 
     @Override
-    protected void createSchemaUri() {
-        schemaName = "SP" + RandomStringUtils.randomAlphanumeric(10);
-        this.uri = this.uri.substring(0, uri.indexOf(';'));
-
-        System.setProperty("spring.datasource.url", uri + ";database=" + schemaName);
+    protected boolean isApplicable() {
+        return "SQL_SERVER".equals(System.getProperty("spring.jpa.database")) //
+                && MATCHER.match(MSSQL_URI_PATTERN, System.getProperty(SPRING_DATASOURCE_URL_KEY)) //
+                && !StringUtils.isEmpty(getSchemaName());
     }
 
     @Override
-    protected boolean isRunningWithSql() {
-        return "SQL_SERVER".equals(System.getProperty("spring.jpa.database"));
+    public void createSchema() {
+        final String uri = System.getProperty(SPRING_DATASOURCE_URL_KEY);
+        final String schemaName = getSchemaName();
+        LOGGER.info("\033[0;33m Creating mssql schema {} if not existing \033[0m", schemaName);
+
+        executeStatement(uri.split(";database=")[0], "CREATE DATABASE IF NOT EXISTS " + schemaName + ";");
```

Comment:
Already mentioned in chat, wrong syntax for `IF NOT EXISTS`, not compatible with MSSQL.